File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int good = 0;
    int goodNodes(TreeNode* root) {
        if(!root) return 0;
        DFS(root, root->val);
        return good;
    }
    void DFS(TreeNode* root, int maxval)
    {
        if(!root) return;
        if(root->val >= maxval) good++;
        maxval = max(maxval, root->val);
        DFS(root->left, maxval);
        DFS(root->right, maxval);
    }
};

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int GoodNodes(TreeNode root) {
        return DFS(root, root.val);
    }
    public int DFS(TreeNode root, int maxval)
    {
        if (root == null) return 0;
        int count = 0;

        if (root.val >= maxval) count++;
        maxval = Math.Max(root.val, maxval);

        count += DFS(root.left, maxval);
        count += DFS(root.right, maxval);

        return count;
    }
}